src/main/scala/beam/sim/BeamSim.scala

```diff
@@ -176,7 +172,15 @@ class BeamSim @Inject()(
   }
 
   override def notifyIterationStarts(event: IterationStartsEvent): Unit = {
+
     beamConfigChangesObservable.notifyChangeToSubscribers()
+
+    beamServices.modeChoiceCalculatorFactory = ModeChoiceCalculator(
```

Comment:
Should this be done here? Wouldn't it make more sense to add it to be a subscriber?

Reply:
to avoid passing any beam specific param in subscriber i have put this code here, If you think its make more sense to move this in subscriber then i can move this in subscriber